pkgs/by-name/go/goldfish/package.nix

```diff
@@ -0,0 +1,90 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  writableTmpDirAsHomeHook,
+
+  buildPackages,
+  windows,
+
+  s7,
+  tbox,
+  isocline,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "goldfish";
+  version = "17.11.20";
+
+  src = fetchFromGitHub {
+    owner = "XmacsLabs";
+    repo = "goldfish";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-eRo7H2z+389kQs0X1XX7nN+uVaSkU2Nr+Q98RoCz6uw=";
+  };
+
+  nativeBuildInputs = with buildPackages; [
+    xmake
+    # make xmake happy
+    (writers.writeBashBin "git" "exit 0")
+    writableTmpDirAsHomeHook
+  ];
+  buildInputs = [
+    s7
+    tbox
+    isocline
+  ]
+  ++ lib.optional stdenv.hostPlatform.isMinGW windows.pthreads;
+
+  env.NIX_LDFLAGS = toString (
+    lib.optionals stdenv.hostPlatform.isMinGW [
+      "-lpthread"
+      "-lws2_32"
+    ]
+  );
+
+  configurePhase = ''
+    runHook preConfigure
+    xmake global --network=private
+    xmake config -m release --yes -vD \
+      --repl=y --ccache=n             \
+      --system-deps=y --pin-deps=n    \
+    ${lib.optionalString stdenv.hostPlatform.isMinGW ''
+      --toolchain=mingw --mingw=${stdenv.cc.outPath}
+    ''}
+    runHook postConfigure
+  '';
+
+  buildPhase = ''
+    runHook preBuild
+    xmake build --yes -j $NIX_BUILD_CORES -vD goldfish
+    runHook postBuild
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # workaround for xmake cannot use system deps
```

Comment:
So it's fine for non-windows platforms?

Reply:
This comment was a bit unclear, I updated it.
Yes, it is fine on non-windows builds